### PR TITLE
Feat(clickhouse): add support for WITH FILL, INTERPOLATED clauses

### LINKED ISSUE #2699: Clickhouse: ORDER BY WITH FILL not working
Hey sqlglot team,

I am trying to integrate SQLglot further in one of my recent open source libraries [SQL Mock](https://github.com/DeepLcom/sql-mock). It is already used for some functionality but I want to extend its usage.

While testing locally, I identified that the Clickhouse `ORDER BY WITH FILL` [syntax](https://clickhouse.com/docs/en/sql-reference/statements/select/order-by#order-by-expr-with-fill-modifier) generates a parsing error:

**Fully reproducible code snippet**

```python
import sqlglot

query = """
SELECT
    toLastDayOfMonth(date) AS observation_month
FROM (
    SELECT
        toStartOfMonth(toDate('2023-01-01') - INTERVAL 1 MONTH) AS date
    ORDER BY
        date WITH FILL
            FROM toDate('2021-01-01')
            STEP INTERVAL 1 MONTH
)
"""

ast = sqlglot.parse_one(query, dialect='clickhouse')
```

Will result in 

```bash
ParseError: Expecting ). Line 8, Col: 10.
   start date we care about and fill the rest from the period_start
        ORDER BY
            date WITH FILL
                FROM toDate('2021-01-01') -- First day we will report on
                STEP 
```

**Official Documentation**
* [ORDER BY Expr WITH FILL Modifier Documentation](https://clickhouse.com/docs/en/sql-reference/statements/select/order-by#order-by-expr-with-fill-modifier)

### PR DESCRIPTION
**Pull request by georgesittas**
Fixes #2699

Reference: https://clickhouse.com/docs/en/sql-reference/statements/select/order-by#order-by-expr-with-fill-modifier

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -3153,7 +3162,21 @@ def _parse_ordered(self, parse_method: t.Optional[t.Callable] = None) -> exp.Ord
         ):
             nulls_first = True
 
-        return self.expression(exp.Ordered, this=this, desc=desc, nulls_first=nulls_first)
+        if self._match_text_seq("WITH", "FILL"):
+            with_fill = self.expression(exp.WithFill)
```
Comment (tobymao) [inline sqlglot/parser.py:3166]: why are you using set instead of just instantiating it?

with_fill = self.expression(..., **{"from": ...

Comment (georgesittas) [reply sqlglot/parser.py:3166]: Ah, yea can do that too. Will update now thanks.

Comment (georgesittas) [reply sqlglot/parser.py]: Addressed

File: sqlglot/parser.py [hunk 2]
```diff
@@ -3153,7 +3162,28 @@ def _parse_ordered(self, parse_method: t.Optional[t.Callable] = None) -> exp.Ord
...
+            if self._match(TokenType.FROM):
+                from_ = self._parse_bitwise()
+            if self._match_text_seq("TO"):
+                to = self._parse_bitwise()
+            if self._match_text_seq("STEP"):
+                step = self._parse_bitwise()
+
+            with_fill = self.expression(
+                exp.WithFill,
+                **{"from": from_, "to": to, "step": step},  # type: ignore
```
Comment (tobymao) [inline sqlglot/parser.py:3179]: just put it all inline? 

match(From) and parse_bitwise() then you don't have to declare

**Pull request merged**